.gitlab-ci.yml

```diff
@@ -1,49 +1,82 @@
-image: ubuntu:22.04
-
 stages:
-  - test
+  - build
+
+.standard-rules:       # Make a hidden job to hold the common rules
+  rules:
+    - if: $CI_PIPELINE_SOURCE == 'merge_request_event'
+    - if: $CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH
+  artifacts:
+    paths:
+      - "build/*/*.tar.gz"
 
 .setup_linux: &setup_linux |
+  echo -e "\e[0Ksection_start:`date +%s`:my_setup_linux[collapsed=true]\r\e[0KHeader of the setup linux collapsible section"
   DEBIAN_FRONTEND=noninteractive
 
-  # set time-zone
-  TZ=Canada/Pacific
-  ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
+  # XXX WHY? set time-zone
+  # TZ=Canada/Pacific
+  # ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
 
   # for downloading
   apt-get update -qq
-  apt-get install -y --no-install-recommends curl gnupg ca-certificates
-
-  # keys used by apt
-  apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 3B4FE6ACC0B21F32
-  apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 40976EAF437D05B5
-  apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 1E9377A2BA9EF27F
+  apt-get install -y --no-install-recommends curl gnupg ca-certificates python3-pip
+  echo -e "\e[0Ksection_end:`date +%s`:my_setup_linux\r\e[0K"
 
 .setup_cpp: &setup_cpp |
-  curl -LJO "https://github.com/aminya/setup-cpp/releases/download/v0.14.2/setup_cpp_linux"
-  chmod +x setup_cpp_linux
-  ./setup_cpp_linux --compiler $compiler --cmake true --ninja true --ccache true --vcpkg true --task true --gcov true
+  echo -e "\e[0Ksection_start:`date +%s`:my_setup_cpp[collapsed=true]\r\e[0KHeader of the setup cpp collapsible section"
+  curl -LJO "https://github.com/aminya/setup-cpp/releases/download/v0.33.0/setup-cpp-x64-linux"
+  chmod +x setup-cpp-x64-linux
+  ./setup-cpp-x64-linux --compiler $compiler --ccache true --vcpkg true --clangtidy true
```

Comment:
These days I recommend using `npm` to install `setup-cpp`. Might be good to update the docs on that repo as well
https://github.com/aminya/setup-cpp#from-terminal